clcache.py

```diff
@@ -748,13 +748,17 @@ def _preprocessToStdout(options):
 
     @staticmethod
     def _parseOptionsAndFiles(cmdline):
-        optionsWithParameter = ['Ob', 'Gs', 'Fa', 'Fd', 'Fm',
-                                'Fp', 'FR', 'doc', 'FA', 'Fe',
-                                'Fo', 'Fr', 'AI', 'FI', 'FU',
-                                'D', 'U', 'I', 'Zp', 'vm',
-                                'MP', 'Tc', 'V', 'wd', 'wo',
-                                'W', 'Yc', 'Yl', 'Tp', 'we',
-                                'Yu', 'Zm', 'F', 'Fi']
+        # sorted by length to handle prefixes
+        optionsWithParameter = [
+            # length 3
+            'doc',
+            # length 2
+            'AI', 'Fa', 'FA', 'Fd', 'Fe', 'Fi', 'FI', 'Fm', 'Fo',
+            'Fp', 'Fr', 'FR', 'FU', 'Gs', 'MP', 'Ob', 'Tc', 'Tp',
+            'vm', 'wd', 'we', 'wo', 'Yc', 'Yl', 'Yu', 'Zm', 'Zp',
+            # length 1
+            'D', 'F', 'I', 'U', 'V', 'W',
+        ]
```

Comment:
Yes, I see basically 2 good ways:

1. Prepare a ready-to use list literal and make sure it is correct (using comments)
2. Define optionsWithParameter as a set because it is the better type here (unsorted, elements contained only once), take it as an implementation detail that the algorithm needs a sorted list and use the sorting you posted.

Both are fine for m, feel free to choose one